axiom/tests/test_log_events.c

```diff
@@ -19,120 +20,138 @@
-  nr_log_events_add_event(events, e, rnd);
-  nr_log_event_destroy(&e);
+  nr_log_event_set_priority(e, 0);
+  nr_log_events_add_event(events, e);
+
+  // minmax heap will store events like this (showing priorities)
+  //         0
+  //       3   2
+  //      1
+  //
+  // so when walking the heap and converting to a vector we expect
+  // the following order:
+  // LOG_MESSAGE_3 (pri 0)
+  // LOG_MESSAGE_0 (pri 3)
+  // LOG_MESSAGE_1 (pri 2)
+  // LOG_MESSAGE_2 (pri 1)
 
   tlib_pass_if_int_equal("events number seen updated", 4,
                          nr_log_events_number_seen(events));
   tlib_pass_if_int_equal("events number saved updated", 4,
                          nr_log_events_number_saved(events));
 
-#define TEST_EVENTS_GET(MESSAGE, IDX)                    \
-  do {                                                   \
-    json = nr_log_events_get_event_json(events, IDX);    \
-    tlib_fail_if_null("no json", json);                  \
-    expected = "[{" \
-                 "\"message\":\"" MESSAGE "\"," \
-                 "\"log.level\":\"" LOG_LEVEL "\"," \
-                 "\"timestamp\":" NR_STR2(LOG_TIMESTAMP) "," \
-                 "\"trace.id\":\"null\"," \
-                 "\"span.id\":\"null\"," \
-                 "\"entity.guid\":\"null\"," \
-                 "\"entity.name\":\"null\"," \
-                 "\"hostname\":\"null\"" \
-              "}]";                                    \
-    tlib_pass_if_str_equal("add event", expected, json); \
+  // convert log event pool to a vector and test
+  vector = nr_vector_create(10, NULL, NULL);
+  nr_log_events_to_vector(events, vector);
+
+#define TEST_EVENTS_GET(MESSAGE, IDX)                              \
+  do {                                                             \
+    pass = nr_vector_get_element(vector, IDX, &test_e);            \
+    tlib_pass_if_true("retrived log element from vector OK", pass, \
+                      "expected TRUE");                            \
+    json = nr_log_event_to_json((nr_log_event_t*)test_e);          \
+    tlib_fail_if_null("no json", json);                            \
+    expected                                                       \
+        = "[{"                                                     \
+          "\"message\":\"" MESSAGE                                 \
+          "\","                                                    \
+          "\"level\":\"" LOG_LEVEL                                 \
+          "\","                                                    \
+          "\"timestamp\":" NR_STR2(LOG_TIMESTAMP) "}]";            \
+    tlib_pass_if_str_equal("add event", expected, json);           \
+    nr_free(json);                                                 \
   } while (0)
 
-  /* Test events are stored in order */
-  TEST_EVENTS_GET(LOG_MESSAGE_0, 0);
-  TEST_EVENTS_GET(LOG_MESSAGE_3, 3);
+  /* Test events are stored in order as explained above for minmax heap*/
```

Comment:
I'm not sure these tests should exists. `nr_minmax_heap_iterate` (underlying implementation of `nr_log_events_to_vector`) documentation says:
> Note : Ordering is not guaranteed; calling code should be prepared to
receive values in any order.

I'm afraid this may make this test unpredictable.

Reply:
The internals of the heap maybe aren't what we care about.
What if I change the test to keep popping the min value and make sure these follow the expected pattern?

Reply:
> ...change the test to keep popping...
Yes or do what you did in 284f848 :) Both solutions will improve test stability.

Reply:
Fixed in commit [7532fdf](https://github.com/newrelic/newrelic-php-agent/pull/515/commits/7532fdf7755f39e1ca31cd646a3192f133f37322).